From 97be65ca2fa7c612c493c2363f608b6b0502b924 Mon Sep 17 00:00:00 2001
From: Piyush Mehta <piyush.mehta@xilinx.com>
Date: Fri, 3 Apr 2020 18:41:51 +0530
Subject: [PATCH 1320/1852] usb: dwc3: otg: fix the function call used to get
 otg descriptor

commit 6b842e3862c94fe05c3a274473b459ede9a6a2e7 from
https://github.com/Xilinx/linux-xlnx.git

This patchs add missing size input arguments used in
__usb_get_extra_descriptor,related to OTG get descriptor.

Signed-off-by: Piyush Mehta <piyush.mehta@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/usb/dwc3/otg.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/drivers/usb/dwc3/otg.c b/drivers/usb/dwc3/otg.c
index e0e2e076335c..bfa954b8aaa6 100644
--- a/drivers/usb/dwc3/otg.c
+++ b/drivers/usb/dwc3/otg.c
@@ -438,7 +438,8 @@ int dwc3_otg_host_release(struct usb_hcd *hcd)
 
 		if (__usb_get_extra_descriptor(udev->rawdescriptors[0],
 				le16_to_cpu(udev->config[0].desc.wTotalLength),
-				USB_DT_OTG, (void **) &desc) == 0) {
+				USB_DT_OTG, (void **)&desc, sizeof(*desc)) ==
+				0) {
 			int err;
 
 			dev_info(&udev->dev, "found OTG descriptor\n");
@@ -1670,7 +1671,7 @@ static int dwc3_otg_notify_connect(struct usb_phy *phy,
 		/* descriptor may appear anywhere in config */
 		err = __usb_get_extra_descriptor(udev->rawdescriptors[0],
 				le16_to_cpu(udev->config[0].desc.wTotalLength),
-				USB_DT_OTG, (void **) &desc);
+				USB_DT_OTG, (void **)&desc, sizeof(*desc));
 		if (err || !(desc->bmAttributes & USB_OTG_HNP))
 			return 0;
 
-- 
2.31.1

